src/argilla/client/feedback/dataset/remote/filtered.py

```diff
@@ -0,0 +1,77 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
+
+from argilla.client.feedback.dataset.remote.base import RemoteFeedbackDatasetBase, RemoteFeedbackRecordsBase
+from argilla.client.feedback.dataset.remote.mixins import ArgillaRecordsMixin
+from argilla.client.feedback.schemas.records import RemoteFeedbackRecord
+
+if TYPE_CHECKING:
+    from uuid import UUID
+
+    import httpx
+
+    from argilla.client.feedback.dataset.remote.dataset import RemoteFeedbackDataset
+    from argilla.client.feedback.schemas.records import FeedbackRecord
+    from argilla.client.feedback.schemas.types import AllowedFieldTypes, AllowedQuestionTypes
+    from argilla.client.workspaces import Workspace
+
+
+class FilteredRemoteFeedbackRecords(RemoteFeedbackRecordsBase, ArgillaRecordsMixin):
+    def __init__(self, dataset: "RemoteFeedbackDataset", filters: Dict[str, Any]) -> None:
+        super().__init__(dataset=dataset)
+
+        self._filters = filters
```

Comment:
Is there any way to move out from the `ArgillaRecordsMixin` all the `self._filters` related operations and keep them as part of the `FilteredRemoteFeedbackRecords` definition? It would be more readable to keep them here. Otherwise, we're coupling both `RemoteFeedbackRecords` and `FilteredRemoteFeedbackRecords` through the logic defined in the  `ArgillaRecordsMixin` class.

Reply:
IMO that was the point on using the mixin, to be able to reuse those, otherwise we would need to redefine the magic methods in the class inheriting from those, and also to define two alternate version of the same code where the only change is an extra arg